runtime/lua/vim/diagnostic.lua

```diff
@@ -228,6 +229,16 @@ end
 --- See |nvim_buf_set_extmark()|.
 --- @field virt_text_hide? boolean
 
+--- @class vim.diagnostic.Opts.VirtualLines
+---
+--- Only show diagnostics for the current line.
+--- (default: `false`)
+--- @field current_line? boolean
```

Comment:
It'd be nice to port this option to the `virtual_text` handler too (in a different PR)

Reply:
Good idea. I'll do it after merging this :)

Reply:
https://github.com/neovim/neovim/pull/32187